public/js/graph/sidebar/sidebar_events.js

```diff
@@ -51,25 +84,28 @@ $('#graph-csc').click(function(e) {
     // activateFeedbackForm();
     // Uncomment to enable graph dragging
     // enableGraphDragging();
-});
+};
 
 
 /**
- * Click function to load Statistics Graph
- * NOTE: A dummy graph is loaded at the moment
+ * Grabs all the graphs currently parsed into the database
+ * @return @returns{JSON[]} A list of JSON objects representing each graph in the database
 **/
-$('#graph-sta').click(function(e) {
+function getGraphsInDatabase() {
     'use strict';
 
-    // Remove current graph
-    $('#graph').empty();
-
-    getRemote('dummygraph.svg');
-
-    buildGraph();
-
-    //setMouseCallbacks();
-
-    //initializeGraphSettings();
-
-});
\ No newline at end of file
+    var graphs;
+    $.ajax({
+        url: 'graphs',
+        dataType: 'json',
+        async: false,
+        success: function (data) {
+            graphs = data;
+        },
+        error: function () {
+            throw 'No graphs in database';
+        }
+    });
+
+    return graphs;
```

Comment:
This is the wrong order in which to do things. You should pass `graphs` as an argument to `createGraphButtons`, calling it inside the `success` function.